packages/dashboard-core-plugins/src/WidgetLoaderPlugin.tsx

```diff
@@ -0,0 +1,94 @@
+import { useMemo, useCallback, type ComponentType, useEffect } from 'react';
+import type { ReactComponentConfig } from '@deephaven/golden-layout';
+import shortid from 'shortid';
+import {
+  assertIsDashboardPluginProps,
+  DashboardPluginComponentProps,
+  DehydratedDashboardPanelProps,
+  PanelEvent,
+  PanelOpenEventDetail,
+  LayoutUtils,
+  useListener,
+} from '@deephaven/dashboard';
+import { usePlugins } from '@deephaven/app-utils';
+import { isWidgetPlugin } from '@deephaven/plugin';
+
+export function WidgetLoaderPlugin(
```

Comment:
JSDocs would be handy for this component. It opens the widget plugin that supports the object when it is opened.